src/tools/runk/libcontainer/src/container.rs

```diff
@@ -34,11 +36,14 @@ pub struct ContainerContext {
     pub spec: Spec,
     pub no_pivot_root: bool,
     pub console_socket: Option<PathBuf>,
+    pub pid_file: Option<PathBuf>,
 }
 
 impl ContainerContext {
-    pub async fn launch(&self, action: ContainerAction, logger: &Logger) -> Result<Pid> {
-        Status::create_dir(&self.state_root, &self.id)?;
+    pub async fn launch(&self, action: ContainerAction, logger: &Logger) -> Result<()> {
```

Comment:
I think the method `launch` should be refactored in this pr or a new pr, because the conditional branch of action is too many here, it's hard to read now and will be unmaintainable in the future.